object/object_test.go

```diff
@@ -0,0 +1,441 @@
+				"metadata": map[string]interface{}{
+					"creationTimestamp": nil,
+				},
+				"spec": map[string]interface{}{},
+				"status": map[string]interface{}{
+					"conditions": []map[string]interface{}{
+						{
+							"type": "WaitingForPlayer",
+						},
+						{
+							"type": "Online",
+						},
+					},
+				},
+			},
+			want: map[string]interface{}{
+				"conditions": []map[string]interface{}{
+					{
+						"type": "WaitingForPlayer",
+					},
+					{
+						"type": "Online",
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range cases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			got, err := GetObjectStatus(tc.obj)
+			if tc.wantErr {
+				assert.Error(t, err)
+				assert.Nil(t, got)
+				return
+			}
+			assert.NoError(t, err)
+			assert.Equal(t, tc.want, got, "result")
+		})
+	}
+}
+
+func TestStatusChanged(t *testing.T) {
+	// Create a scheme with testdata scheme info.
+	scheme := runtime.NewScheme()
+	assert.Nil(t, tdv1alpha1.AddToScheme(scheme))
+
+	cases := []struct {
+		name    string
+		oldo    runtime.Object
+		newo    runtime.Object
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "no status",
+			oldo: &tdv1alpha1.Game{},
+			newo: &tdv1alpha1.Game{},
+			want: false,
+		},
+		{
+			name: "no change",
+			oldo: &tdv1alpha1.Game{
+				Status: tdv1alpha1.GameStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type: "WaitingForPlayer",
+						},
+					},
+				},
+			},
+			newo: &tdv1alpha1.Game{
+				Status: tdv1alpha1.GameStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type: "WaitingForPlayer",
+						},
+					},
+				},
+			},
+			want: false,
+		},
+		{
+			name: "status added",
+			oldo: &tdv1alpha1.Game{},
+			newo: &tdv1alpha1.Game{
+				Status: tdv1alpha1.GameStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type: "Booting",
+						},
+					},
+				},
+			},
+			want: true,
+		},
+		{
+			name: "status removed",
+			oldo: &tdv1alpha1.Game{
+				Status: tdv1alpha1.GameStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type: "SelfDestructing",
+						},
+					},
+				},
+			},
+			newo: &tdv1alpha1.Game{},
+			want: true,
+		},
+		{
+			name: "add additional",
+			oldo: &tdv1alpha1.Game{
+				Status: tdv1alpha1.GameStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type: "WaitingForPlayer",
+						},
+					},
+				},
+			},
+			newo: &tdv1alpha1.Game{
+				Status: tdv1alpha1.GameStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type: "WaitingForPlayer",
+						},
+						{
+							Type: "InstallingUpdate",
+						},
+					},
+				},
+			},
+			want: true,
+		},
+	}
+	for _, tc := range cases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			got, err := StatusChanged(scheme, tc.oldo, tc.newo)
+			if tc.wantErr {
+				assert.Error(t, err)
+				assert.Nil(t, got)
+				return
+			}
+			assert.NoError(t, err)
+			assert.Equal(t, tc.want, got, "result")
+		})
+	}
+}
+
+func TestNestedFieldNoCopy(t *testing.T) {
```

Comment:
Nice to have test coverage for this.
Maybe we can upstream these tests in the future to the kubebuilder-declarative-pattern repo.